**1) Tulis fungsi Python untuk menemukan angka terbesar dari tiga angka.**

**Jawab**
* Berikut ini merupakan fungsi untuk menemukan angka terbesar dari sebuah list yang ada.
* list nilai yang diperoleh dari nilai "arr"
* dengan kondisi, untuk setiap elemen didalam range nilai 1 hingga panjang elemen dari array, 
   jika array elemen i lebih besar dari elemen Max, 
   maka elemen max (imax) = arr elemen i
* nilai dikembalikan ke imax sesuai list nilai yang ada. 
* fungsi ini berlaku tidak hanya memberikan list sebanyak tiga angka, namun bisa lebih. 

In [450]:
def num_Max(arr):
    iMax = arr[0]
    for i in range(1, len(arr)):
        if(arr[i] > iMax):
            iMax = arr[i]
    return iMax
arr = [4,18,9]
iMax = arr[0]
for i in range(1, len(arr)):
    if(arr[i] > iMax):
        iMax = arr[i]
print (iMax)

18


* Nilai maksimum dari elemen list yang diberikan adalah : 18, dan akan berubah sesuai list yang diberikan

**2) Tulis fungsi Python untuk menjumlahkan semua angka dalam list.**

**Jawab**
* fungsi python berikut akan menjumlahkan semua list yang diberikan.
* dimana menggunakan non keyword argument apabila list yang dimasukkan lebih dari 1.
* dimana, c adalah argumen ke satu.  
* untuk setiap elemen pada non keyword argument, nilai c merupakan jumlah sama dengan nilai elemen tersebut. 


In [81]:
def jumlah (a,*b):
    c = a
    for i in b:
        c += i 
    print (c)

In [168]:
jumlah (1,3,4,5,7,9,11)

40


* Jumlah semua angka dari list tersebut adalah : 40, dan akan berubah sesuai list yang diberikan 

**3) Tulis fungsi Python untuk mengalikan semua angka dalam list.**
     Sample list : [8, 2, 3, -1, 7], Output yang Diharapkan : -336

**Jawab** 
* merupakan fungsi untuk mengalikan nilai yang ada di dalam list.
* digunakan nilai c sama dengan kali dengan elemen nya. 
* hal ini hampir sama dengan pernyataan pada soal nomor 2, namun pada kasus ini operator nya adalah kali. 


In [83]:
def new (a,*b):
    c = a
    for i in b:
        c *= i 
    print (c)

In [70]:
new (2,5,7,9)

630


* Jumlah hasil kali dari setiap elemen list adalah: 630, dan akan berubah sesuai list yang diberikan. 

**4) Tulis program Python untuk membalikkan string.** Contoh String : "1234abcd", Output yang Diharapkan : "dcba4321"

**Jawab**
* ini merupakan fungsi unutk membalikkan nilai string yang diberikan. 
* untuk menghasilkan nilai yang terbalik ketika fungsi dijalankan adalah menggunakan metode slicing. 
* dimana hasil adalah dengan semua indeks, dan dihitung dari indeks paling belakang dari sebuah string. 


In [172]:
def reverse_(string):
    result = string [::-1]
    print (result)

In [173]:
reverse_("Hallo Annisa")

asinnA ollaH


* Hasil dapat berubah sesuai dengan string yang diberikan 

**5) Tulis fungsi Python yang menerima string dan menghitung jumlah
huruf besar dan huruf kecil.** Contoh String : 'Hello World' , Output yang Diharapkan : Jumlah
Karakter Huruf Besar : 2 dan Jumlah Karakter Huruf Kecil : 8

**Jawab**
* fungsi ini berlaku untuk data jenis string, tidak dapat digunakan jenis integer. 
* dimana result1 merupakan jumlah dari pemeriksaan setiap item berhurup kapital. 
* dimana result2 merupakan jumlah dari pemeriksaan setiap item berhuruf kecil. 
* menggunakan fungsi map sebagai u=fungsi untuk memeriksa setiap item didalam object atau kalimat. 

In [454]:
def n_chars(string):
    result1 = sum (map(str.isupper, string))
    result2 = sum (map(str.islower, string))
    print ('Jumlah huruf kapital:',result1, 'Jumlah huruf kecil:', result2)

In [453]:
n_chars ('Ayam Bebek Angsa dipotong')

Jumlah huruf kapital: 3 Jumlah huruf kecil: 19


* Jumlah huruf kapital dan kecil akan berubah sesuai kalimat yang diberikan. 

**6) Tulis fungsi Python yang mengambil daftar dan mengembalikan list
baru dengan elemen unik dari list pertama.**
Sample list : [1,2,3,3,3,3,4,5], Unique list : [1, 2, 3, 4, 5]

**Jawab**
* fungsi ini merupakan fungsi untuk mendapatkan nilai unik dari sebuah list yang ada. 
* nilai diurutkan dari nilai terkecil hingga nilai terbesar. 
* setiap nilai ganda akan diambil satu angka untuk disimpan ke dalam list value baru. 

In [161]:
def unik(list):
    list = np.array([list])
    result = np.unique(list)
    print (result)
    

In [166]:
unik ([1,1,13,3,5,66,5,0,0,0,5555,7])

[   0    1    3    5    7   13   66 5555]


* Hasil akan berubah sesuai nilai list yang diberikan. 


**7) Tulis fungsi Python yang menggunakan angka sebagai input
parameter dan periksa angkanya prima atau tidak.**

**Jawab**
* fungsi ini mengecek sebuang nilai atau angka merupakan bilangan prima atau bukan.
* dimana nilai yang diberikan merupakan nilai input parameter oleh user sendiri. 
* dengan kondisi, untuk setiap elemen input didalam range nilai 2 dan nilai elemen ke -n, 
  jika elemen n dengan pembagi elemen habis dibagi dengan elemen adalah sama dengan nol, 
  berarti elemen terebut bukan elemen bilangan prima. 
* jika elemen habis dibagi tidak sama dengan nol, maka elemen termasuk ke dalam bilangan prima. 


In [456]:
def prima (n):
    n = int(input("angka: "))
    for i in range (2, n):
        if n % i == 0: 
            return False
    return True
prima(n)

angka: 97


True

* Hasil True menyatakan bahwa nilai atau angka yang dimasukkan merupakan bilangan prima. 
* hasil False menyatakan bahwa nilai atau angka yang dimasukkan merupakan bukan bilangan prima. 


**8) Tulis program Python untuk mencetak bilangan genap dari list yang
diberikan.**
Sample list : [1, 2, 3, 4, 5, 6, 7, 8, 9], Hasil yang Diharapkan : [2, 4, 6, 8]

**Jawab**
* merupakan fungsi untuk menampilkan bilangan genap dari nilai yang diberikan. 
* menggunaka non keywords argumen sebagai pilihan karena untuk membebaskan nilai 
  yang diberikan lebih dari satu. 
* dengan menggunakan kondisi for, unutk setiap elemen didalam nilai y, 
  berlaku jika elemen habis dibagi dengan 2 adalah 0, maka merupakan bilangan genap. 

In [445]:
def bilangan (*y):
    result = []
    for i in (y):
        if i % 2==0:
            print (i)
            result= True
    

In [448]:
bilangan (1,3,4,6,8,12,16,4,11)

4
6
8
12
16
4


* Hasil dapat berubah sesuai nilai yang diberikan 


**FINISHED**